### PR TITLE
feat(gossip-menu-option-preview): add preview in the gossip menu option editor

### PR DESCRIPTION
**Pull request by Helias**

### CODE REVIEW

File: src/app/features/gossip/gossip-menu-option-preview/gossip-menu-option-preview.component.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import { OPTION_IMG } from '@keira-constants/options/gossip-option-icon';
+
+@Component({
+  selector: 'keira-gossip-menu-option-preview',
+  templateUrl: './gossip-menu-option-preview.component.html',
+  styleUrls: ['./gossip-menu-option-preview.component.scss']
+})
+export class GossipMenuOptionPreviewComponent {
+  @Input() options: GossipMenuOption[];
+  @Input() show: boolean = true;
```
Comment (FrancescoBorzi) [inline src/app/features/gossip/gossip-menu-option-preview/gossip-menu-option-preview.component.ts:13]: Unnecessary type, it's inferred

File: src/app/features/gossip/gossip-menu-option-preview/gossip-menu-option-preview.module.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { NgModule } from '@angular/core';
+import { BrowserModule } from '@angular/platform-browser';
+
+import { QueryOutputModule } from '@keira-shared/modules/query-output/query-output.module';
+import { GossipMenuOptionPreviewComponent } from './gossip-menu-option-preview.component';
+import { PerfectScrollbarModule } from 'ngx-perfect-scrollbar';
```
Comment (FrancescoBorzi) [inline src/app/features/gossip/gossip-menu-option-preview/gossip-menu-option-preview.module.ts:6]: Move up

File: src/app/features/gossip/gossip-menu/gossip-menu.integration.spec.ts [hunk 3]
```diff
@@ -99,7 +99,7 @@ describe('GossipMenu integration tests', () => {
       page.clickExecuteQuery();
 
       page.expectDiffQueryToContain(expectedQuery);
-      expect(querySpy).toHaveBeenCalledTimes(1);
+      expect(querySpy).toHaveBeenCalled();
```
Comment (FrancescoBorzi) [inline src/app/features/gossip/gossip-menu/gossip-menu.integration.spec.ts:102]: Specify the correct amount of calls

File: src/app/features/gossip/select-gossip/select-gossip.integration.spec.ts [hunk 4]
```diff
@@ -135,7 +135,7 @@ describe('SelectGossip integration tests', () => {
 
       page.clickElement(page.searchBtn);
 
-      expect(querySpy).toHaveBeenCalledTimes(1);
+      expect(querySpy).toHaveBeenCalled();
```
Comment (FrancescoBorzi) [inline src/app/features/gossip/select-gossip/select-gossip.integration.spec.ts:138]: Same

File: src/app/shared/modules/selectors/npc-text-selector/npc-text-selector-btn.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { NpcTextSelectorBtnComponent } from './npc-text-selector-btn.component';
+import { NpcTextSelectorModule } from './npc-text-selector.module';
+
+describe('NpcTextSelectorBtnComponent', () => {
+  let component: NpcTextSelectorBtnComponent;
+  let fixture: ComponentFixture<NpcTextSelectorBtnComponent>;
```
Comment (FrancescoBorzi) [inline src/app/shared/modules/selectors/npc-text-selector/npc-text-selector-btn.component.spec.ts:8]: Avoid shared variables, use the setup function

File: src/app/shared/modules/selectors/npc-text-selector/npc-text-selector-modal.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+import { NpcTextSelectorModalComponent } from './npc-text-selector-modal.component';
+import { MysqlQueryService } from '../../../services/mysql-query.service';
+import { MockedMysqlQueryService } from '@keira-testing/mocks';
+import { NpcTextSearchService } from '../../search/npc-text-search.service';
+import { NpcTextSelectorModule } from './npc-text-selector.module';
+
+describe('ItemSelectorModalComponent', () => {
+  let component: NpcTextSelectorModalComponent;
+  let fixture: ComponentFixture<NpcTextSelectorModalComponent>;
+  let searchService: NpcTextSearchService;
```
Comment (FrancescoBorzi) [inline src/app/shared/modules/selectors/npc-text-selector/npc-text-selector-modal.component.spec.ts:14]: Same

File: src/app/shared/modules/selectors/npc-text-selector/npc-text-selector.module.ts [hunk 7]
```diff
@@ -0,0 +1,35 @@
+import { NgModule } from '@angular/core';
+import { BrowserModule } from '@angular/platform-browser';
+import { ReactiveFormsModule } from '@angular/forms';
+
+import { NpcTextSelectorModalComponent } from './npc-text-selector-modal.component';
+import { NpcTextSelectorBtnComponent } from './npc-text-selector-btn.component';
+import { highlightOptions } from '@keira-config/highlight.config';
+import { HighlightModule } from 'ngx-highlightjs';
+import { NgxDatatableModule } from '@swimlane/ngx-datatable';
+import { ModalModule } from 'ngx-bootstrap/modal';
```
Comment (FrancescoBorzi) [inline src/app/shared/modules/selectors/npc-text-selector/npc-text-selector.module.ts:10]: Move libs up

File: src/app/features/gossip/gossip-menu/gossip-menu.integration.spec.ts (hunk 3 above)
Comment (Helias) [reply src/app/features/gossip/gossip-menu/gossip-menu.integration.spec.ts:102]: it depends on the number of rows, we do not have an array which contains all of them to get the .length, moreover, it depends also if create new is true or not
I think that for this time we can skip specifying the number of times

**Pull request merged**